src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -121,6 +135,21 @@ private static void RequestCallback(
             }
         }
 
+        private static void OnRequestConnectedToServer(WinHttpRequestState state, IPAddress connectedIPAddress)
+        {
+            Debug.Assert(state != null);
+            Debug.Assert(state.Handler != null);
+
+            if (state.Handler.TryRemoveCertificateFromCache(connectedIPAddress))
```

Comment:
I don't think we need to clean the cache here. As long as the certificate is same as it was before and we connections to the same TargetHost. Certainly for HTTP/1.1 I would expect many connections to the same host if there are parallel request.